internal/controller/etcdopstask/reconcile_delete.go

```diff
@@ -0,0 +1,146 @@
+package etcdopstask
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/etcd-druid/api/core/v1alpha1"
+	ctrlutils "github.com/gardener/etcd-druid/internal/controller/utils"
+	druiderr "github.com/gardener/etcd-druid/internal/errors"
+	opsTask "github.com/gardener/etcd-druid/internal/task"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// triggerDeletionFlow handles the deletion flow for EtcdOpsTask resources.
+func (r *Reconciler) triggerDeletionFlow(ctx context.Context, taskHandler opsTask.Handler, task *v1alpha1.EtcdOpsTask) ctrlutils.ReconcileStepResult {
+	logger := taskHandler.Logger()
+	logger.Info("Triggering deletion flow", "completed", task.IsCompleted(), "markedForDeletion", task.IsMarkedForDeletion())
+
+	// If task is completed but not marked for deletion, wait for TTL expiry
+	if task.IsCompleted() {
+
+		if !task.HasTTLExpired() {
+			logger.Info("Task completed but TTL not expired yet, will requeue after TTL", "ttlSeconds", task.Spec.TTLSecondsAfterFinished)
+			return ctrlutils.ReconcileAfter(task.GetTimeToExpiry(), "Task completed, waiting for TTL to expire")
+		}
+		logger.Info("Task TTL expired, proceeding with deletion")
+	}
+
+	deletionStepFns := []StepFunction{
+		{
+			StepName: "CleanupTaskResources",
```

Comment:
There is no need for `StepName`, please remove it. Each step should produce a log.